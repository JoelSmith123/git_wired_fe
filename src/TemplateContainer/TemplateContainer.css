.TemplateContainer {
  background-color: #fff;
  height: 800px;
  margin: 50px auto;
  width: 80%;
}