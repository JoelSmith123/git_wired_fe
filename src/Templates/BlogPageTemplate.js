import React, { Component } from 'react';
import './Templates.css';

export default class BlogPageTemplate extends Component {
  constructor() {
    super()

    this.state = {
      headerButtons: 0,
      templateObj: {}
    }
  }

  componentDidUpdate = () => { 
    this.findElementChildrenToHide(this.refs.BlogPageTemplate)  
    this.props.buildElementObjectForBackend(this.state.templateObj)
  }

  findElementChildrenToHide = (element) => {
    if (element.children) {
      Object.values(element.children).forEach(child => {
        if (this.props.cardIds) {
          if (this.props.cardIds.includes(parseInt(child.id))) {
            child.classList.add('hideTemplateElement')
          } else {
            child.classList.remove('hideTemplateElement')            
          }            
        } else {
          child.classList.remove('hideTemplateElement')                      
        }

        this.findElementChildrenToHide(child)
      })
    }      
  }

  addBtnToHeader = (event) => {
    event.preventDefault()

    this.setState({ headerButtons: this.state.headerButtons + 1 })
  }

  renderBtnToHeader = () => {
    const btnArr = []
    for (let i = 0; this.state.headerButtons > i; i++) {
      btnArr.push(<button key={i} className='template-header-btn'></button>)
    }
    return btnArr
  }


  // {
  //   ref:           "BlogPageTemplate", 
  //   header:        { label: 'label', desc: 'longer description', card: 0 },
  //   headerButton1: { label: 'label', desc: 'longer description', card: 0 },
  //   headerButton2: { label: 'label', desc: 'longer description', card: 0 },
  //   headerButton3: { label: 'label', desc: 'longer description', card: 0 },
  //   pageSection1:  { label: 'label', desc: 'longer description', card: 0 },  
  //   pageSection2:  { label: 'label', desc: 'longer description', card: 0 },  
  //   footer:        { label: 'label', desc: 'longer description', card: 0 }  
  // }


  render() {
    return (
      <div className='BlogPageTemplate' ref='BlogPageTemplate'>
        <div className='template-header' id='11'>
        <div className='pseudo-btn-space'></div>
          <i className="far fa-plus-square" onClick={this.addBtnToHeader}></i>
          {
            this.renderBtnToHeader()
          }
        </div>
        <div className='template-card-container'>
          <div className='template-page-section' id='12'></div>
          <div className='template-page-section' id='13'></div>        
        </div>
        <div className='template-footer' id='14'></div>
      </div>
    )
  }
}
