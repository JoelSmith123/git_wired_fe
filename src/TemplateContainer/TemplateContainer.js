import React, { Component } from 'react';
import './TemplateContainer.css';

export default class TemplateContainer extends Component {
  constructor() {
    super()
  }

  render() {
    return (
      <div className='TemplateContainer'>

      </div>
    )
  }
}