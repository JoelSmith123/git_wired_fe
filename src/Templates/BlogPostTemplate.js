import React, { Component } from 'react';
import './Templates.css';

export default class BlogPostTemplate extends Component {
  constructor() {
    super()

    this.state = {
      headerButtons: 0
    }
  }

  addBtnToHeader = (event) => {
    event.preventDefault()
    this.setState({ headerButtons: this.state.headerButtons + 1 })
  }

  renderBtnToHeader = () => {
    const btnArr = []
    for (let i = 0; this.state.headerButtons > i; i++) {
      btnArr.push(<button key={i} className='template-header-btn'></button>)
    }
    return btnArr
  }

  render() {
    return (
      <div className='BlogPostTemplate'>
        <div className='template-header'>
        <div className='pseudo-btn-space'></div>
          <i className="far fa-plus-square" onClick={this.addBtnToHeader}></i>
          {
            this.renderBtnToHeader()
          }
        </div>
        <div className='template-card-container'>
          
        </div>
        <div className='template-footer'></div>
      </div>
    )
  }
}
