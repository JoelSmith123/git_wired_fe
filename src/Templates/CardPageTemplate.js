import React, { Component } from 'react';
import './Templates.css';

export default class CardPageTemplate extends Component {
  constructor() {
    super()
  }

  render() {
    return (
      <div className='CardPageTemplate'>
        <div className='template-header'></div>
        <div className='template-card-container'>
          <div className='template-card'></div>
          <div className='template-card'></div>
          <div className='template-card'></div>
          <div className='template-card'></div>
        </div>
        <div className='template-footer'></div>
      </div>
    )
  }
}