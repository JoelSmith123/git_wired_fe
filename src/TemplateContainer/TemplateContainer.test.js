import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import TemplateContainer from './TemplateContainer.js'
import CardPageTemplate from '../Templates/CardPageTemplate.js';
import BlogPageTemplate from '../Templates/BlogPageTemplate.js';
import BlogPostTemplate from '../Templates/BlogPostTemplate.js';

describe('TemplateContainer', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<TemplateContainer />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match snapshot', () => {
    const wrapper = shallow(<TemplateContainer />)
    expect(wrapper).toMatchSnapshot()
  })    

  it('updatesChildrenState', () => {
    const mockCardIds = [10, 9, 8]
    const wrapper = shallow(<TemplateContainer cardIds={mockCardIds}/>)
    
    expect(wrapper.instance().updateChildrenState()).toEqual(mockCardIds)
  })

  it('renders card page template', () => {
    const wrapper = shallow(<TemplateContainer />)
    
    expect(wrapper.instance().renderCurrentTemplate('card-page-template')).toEqual(<CardPageTemplate />)
  })

  it('renders blog page template', () => {
    const wrapper = shallow(<TemplateContainer />)
    
    expect(wrapper.instance().renderCurrentTemplate('blog-page-template')).toEqual(<BlogPageTemplate />)
  })

  xit('renders the right template', () => {
    // TO DO - TEST ME
  })


});
