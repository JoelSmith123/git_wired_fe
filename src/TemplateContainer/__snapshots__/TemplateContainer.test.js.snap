// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TemplateContainer should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TemplateContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h2>
        Please select a template type from the Recent Projects dropdown menu
      </h2>,
      "className": "TemplateContainer",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Please select a template type from the Recent Projects dropdown menu",
      },
      "ref": null,
      "rendered": "Please select a template type from the Recent Projects dropdown menu",
      "type": "h2",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h2>
          Please select a template type from the Recent Projects dropdown menu
        </h2>,
        "className": "TemplateContainer",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Please select a template type from the Recent Projects dropdown menu",
        },
        "ref": null,
        "rendered": "Please select a template type from the Recent Projects dropdown menu",
        "type": "h2",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
